mnist/model-train.yaml

```diff
@@ -112,147 +112,145 @@ spec:
-                          key: awsSecretAccessKey
-                    - name: AWS_DEFAULT_REGION
-                      value: {{workflow.parameters.aws-region}}
-                    - name: AWS_REGION
-                      value: {{workflow.parameters.aws-region}}
-                    - name: S3_REGION
-                      value: {{workflow.parameters.aws-region}}
-                    - name: S3_USE_HTTPS
-                      value: "{{workflow.parameters.s3-use-https}}"
-                    - name: S3_VERIFY_SSL
-                      value: "{{workflow.parameters.s3-verify-ssl}}"
-                    - name: S3_ENDPOINT
-                      value: {{workflow.parameters.s3-endpoint}}
-                restartPolicy: OnFailure
-          - replicas: {{workflow.parameters.tf-ps}}
-            tfReplicaType: PS
-            template:
-              spec:
-                containers:
-                  - image: {{workflow.parameters.tf-model-image}}
-                    name: tensorflow
-                    imagePullPolicy: Always
-                    env:
-                    - name: TF_MODEL_DIR
-                      value: {{inputs.parameters.s3-model-url}}
-                    - name: TF_EXPORT_DIR
-                      value: {{workflow.parameters.model-name}}
-                    - name: TF_TRAIN_STEPS
-                      value: "{{workflow.parameters.model-train-steps}}"
-                    - name: TF_TF_BATCH_SIZE
-                      value: "{{workflow.parameters.model-batch-size}}"
-                    - name: TF_LEARNING_RATE
-                      value: "{{workflow.parameters.model-learning-rate}}"
-                    - name: AWS_ACCESS_KEY_ID
-                      valueFrom:
-                        secretKeyRef:
-                          name: {{workflow.parameters.aws-secret}}
-                          key: awsAccessKeyID
-                    - name: AWS_SECRET_ACCESS_KEY
-                      valueFrom:
-                        secretKeyRef:
-                          name: {{workflow.parameters.aws-secret}}
-                          key: awsSecretAccessKey
-                    - name: AWS_DEFAULT_REGION
-                      value: {{workflow.parameters.aws-region}}
-                    - name: AWS_REGION
-                      value: {{workflow.parameters.aws-region}}
-                    - name: S3_REGION
-                      value: {{workflow.parameters.aws-region}}
-                    - name: S3_USE_HTTPS
-                      value: "{{workflow.parameters.s3-use-https}}"
-                    - name: S3_VERIFY_SSL
-                      value: "{{workflow.parameters.s3-verify-ssl}}"
-                    - name: S3_ENDPOINT
-                      value: {{workflow.parameters.s3-endpoint}}
-                restartPolicy: OnFailure
-          terminationPolicy:
-            chief:
-              replicaIndex: 0
-              replicaName: MASTER
+          cleanPodPolicy: All
```

Comment:
I would recommend not setting cleanPodPolicy; this will use the default which only delete running pods so completed pods are left alone to preserve the logs.

Reply:
@jlewi Thx, I removed it